aspects/schema.go

```diff
@@ -121,17 +144,29 @@ func (s *StorageSchema) parse(raw json.RawMessage) (parser, error) {
 }
 
 func (s *StorageSchema) newTypeSchema(typ string) (parser, error) {
-	// TODO: add any, int, number, bool, array and user-defined types
+	// TODO: add any, int, number, bool and array
 	switch typ {
 	case "map":
 		return &mapSchema{topSchema: s}, nil
 	case "string":
 		return &stringSchema{}, nil
 	default:
+		if typ != "" && typ[0] == '$' {
+			return s.getUserType(typ[1:])
+		}
+
 		return nil, fmt.Errorf("cannot parse unknown type %q", typ)
 	}
 }
 
+func (s *StorageSchema) getUserType(ref string) (parser, error) {
```

Comment:
something like:

```
    if ref == "" || ref[0] != '$' {
        return nil, nil
    }
```

another approach would be to return a errNotATypeRef error, anyway even returning nil, nil might be alright here as the helper is unexported